addons/mail/static/src/core/common/thread_model.js

```diff
@@ -257,6 +246,46 @@ export class Thread extends Record {
     custom_notifications = false;
     /** @type {String} */
     mute_until_dt;
+    discussCategory = Record.one("DiscussAppCategory", {
+        compute() {
+            return this._computeDiscussCategory();
+        },
+        inverse: "threads",
+    });
+
+    /**
+     * @returns {object} The  discuss category this thread belongs to. Used to
+     * group threads by category in the discuss app sidebar.
+     */
+    _computeDiscussCategory() {
+        if (this.type === "channel") {
+            return {
+                extraClass: "o-mail-DiscussSidebarCategory-channel",
+                id: "channels",
+                name: _t("Channels"),
+                isOpen: false,
+                canView: true,
+                canAdd: true,
+                serverStateKey: "is_discuss_sidebar_category_channel_open",
+                addTitle: _t("Add or join a channel"),
+                addHotkey: "c",
+            };
+        }
+        if (["group", "chat"].includes(this.type)) {
+            return {
+                extraClass: "o-mail-DiscussSidebarCategory-chat",
+                id: "chats",
+                name: _t("Direct messages"),
+                isOpen: false,
+                canView: false,
+                canAdd: true,
+                serverStateKey: "is_discuss_sidebar_category_chat_open",
+                addTitle: _t("Start a conversation"),
+                addHotkey: "d",
+            };
+        }
+        return null;
+    }
```

Comment:
Passing all the data of a category is not of the responsibility of `ThreadModel`.
It would be better to assign them once in `DiscussCategory` model or `DiscussAppModel`.

This is a computed relational field, you may want to reduce processing by just providing `{ id }` in the data... or use `this._store.discuss.[category]` (the record itself), which is also pretty cheap.